### ISSUE TITLE
dakota.run() crashes when run in a test loop

### ISSUE THREAD

**Issue by kbarnhart** [csdms/dakota#56]
I have used dakotathon to create a looping script that launches a series of dakota runs.

As a test, I ran this script with 6 iterations of the loop and the analysis_driver file modified so that dakota just created output files and running the model was skipped. Each iteration asks dakota to create a 100 sample experiment (so 600 run folders would be created). 

Half way through running, this effort fully crashes my computer requiring a restart. 

If I modify the script to initialize the dakota object, create the dakota.in file, and not execute dakota.run(), and then I run each of the six dakota.in files from the command line, I have no problems and all of the dakota runs complete in less time that it takes for my computer to crash.

I've been able to fix this problem by replacing the original run function with the following: 

```python
def run(self):
        """Run the Dakota experiment.

        Run is executed in the directory specified by run_directory keyword and
        run log and error log are created.
        """
        os.chdir(self.run_directory)

        with open(self.run_log, "w") as file_out:
            with open(self.error_log, "w") as error_out:
                subprocess.call(['dakota',
                                 '-i', self.input_file,
                                 '-o', self.output_file],
                                stdout=file_out,
                                stderr=error_out)
```

Here I've also added functionality for:
1) dynamic naming of the run directory
2) dynamic naming of a stdout file
3) dynamic naming of a stderr file

The stdout file provides the same functionality as the standard ```&> run.log``` call in a dakota command line call such as:
```dakota -i dakota.in -o dakota.out &> run.log```

This change to run() fixes the crashing problem on my computer. 

The code I've used is pretty extensive, let me know if you'd like me to pare it down to provide a minimally complete example that reproduces this problem.

**Comment by mdpiper** [comment]
@kbarnhart Great job tracking down and fixing this problem!

**Comment by mdpiper** [comment]
Fixed with #57.
